File: src/main/java/ssg/search/constant/Cls.java
package ssg.search.constant;

public enum Cls {
	// 이마트 점포상품
    EMART{
        public String getPrefix() {
            return "EM";
        }
		public String getField(){
			return null;
		}
    },
    // 오반장 상품
    OBANJANG{
        public String getPrefix() {
            return "OBANJANG";
        }
        public String getField(){
			return null;
		}
    },
    // 신문광고 상품
    NEWS{
        public String getPrefix() {
            return "NEWS";
        }
        public String getField(){
			return null;
		}
    },
    // 특가상품
    SP_PRICE{
        public String getPrefix() {
            return "SPPRICE";
        }
        public String getField(){
			return null;
		}
    },
    // 이마트 온라인 상품 ( SITE, SALESTR_LST 를 사용함 )
    EMARTONLINE{
        public String getPrefix() {
            return "EMON";
        }
        public String getField(){
			return null;
		}
    },
    // 트레이더스 상품 ( SITE, SALESTR_LST 를 사용함 )
    TRADERS{
        public String getPrefix() {
            return "TR";
        }
        public String getField(){
			return null;
		}
    },
    // 분스 상품 ( SITE, SALESTR_LST 를 사용함 )
    BOONS{
        public String getPrefix() {
            return "BN";
        }
        public String getField(){
			return null;
		}
    },
    // 신세게 몰 상품 ( SITE, SALESTR_LST 를 사용함 )
    SHINSEGAE{
        public String getPrefix() {
            return "SM";
        }
        public String getField(){
			return null;
		}
    },
    // 신세계 백화점 상품 ( SITE, SALESTR_LST 를 사용함 )
    DEPARTMENT{
        public String getPrefix() {
            return "SD";
        }
        public String getField(){
			return null;
		}
    };
    public abstract String getPrefix();
    public abstract String getField();
}

File: src/main/kotlin/ssg/front/search/api/core/constants/Cls.kt
package ssg.front.search.api.core.constants

enum class Cls {
    // 이마트 점포상품
    EMART {
        override val prefix: String
            get() = "EM"
    },
    // 오반장 상품
    OBANJANG {
        override val prefix: String
            get() = "OBANJANG"
    },
    // 신문광고 상품
    NEWS {
        override val prefix: String
            get() = "NEWS"
    },
    // 특가상품
    SP_PRICE {
        override val prefix: String
            get() = "SPPRICE"
    },
    // 이마트 온라인 상품 ( SITE, SALESTR_LST 를 사용함 )
    EMARTONLINE {
        override val prefix: String
            get() = "EMON"
    },
    // 트레이더스 상품 ( SITE, SALESTR_LST 를 사용함 )
    TRADERS {
        override val prefix: String
            get() = "TR"
    },
    // 분스 상품 ( SITE, SALESTR_LST 를 사용함 )
    BOONS {
        override val prefix: String
            get() = "BN"
    },
    // 신세게 몰 상품 ( SITE, SALESTR_LST 를 사용함 )
    SHINSEGAE {
        override val prefix: String
            get() = "SM"
    },
    // 신세계 백화점 상품 ( SITE, SALESTR_LST 를 사용함 )
    DEPARTMENT {
        override val prefix: String
            get() = "SD"
    };

    abstract val prefix: String
}
